freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/user/model/User.java

```diff
@@ -33,15 +33,70 @@ public String getName() {
         return name;
     }
 
+    public void setName(String name) {
+        this.name = name;
+    }
+
     public String getFirstName() {
         return firstName;
     }
 
+    public void setFirstName(String firstName) {
+        this.firstName = firstName;
+    }
+
     public String getLastName() {
         return lastName;
     }
 
+    public void setLastName(String lastName) {
+        this.lastName = lastName;
+    }
+
     public Set<String> getGroups() {
         return groups;
     }
+
+    public void setGroups(Set<String> groups) {
+        this.groups = groups;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        User user = (User) o;
+
+        if (!name.equals(user.name)) {
+            return false;
+        }
+        if (firstName != null ? !firstName.equals(user.firstName) : user.firstName != null) {
+            return false;
+        }
+        return lastName != null ? lastName.equals(user.lastName) : user.lastName == null;
+
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
I think it's much readable to use Objects.hash or HashCodeBuilder from Apache commons

Reply:
done